libcudacxx/include/cuda/std/detail/libcxx/include/barrier

```diff
@@ -384,9 +382,8 @@ private:
     static _LIBCUDACXX_INLINE_VISIBILITY constexpr
     uint64_t __init(ptrdiff_t __count) noexcept
     {
-#if (_LIBCUDACXX_DEBUG_LEVEL >= 2)
-        _LIBCUDACXX_DEBUG_ASSERT(__count >= 0);
-#endif
+        // We would like to assert that __count >= 0 in debug mode, but this
+        // fails to compile due to constexpr.
```

Comment:
question: `_LIBCUDACXX_DEBUG_ASSERT` fails to compile?

suggestion: If it only fails for older compilers like gcc-6 we can simply ifdef it out for those older compilers and leave it for the newer ones